core/controller/src/main/scala/whisk/core/controller/ApiUtils.scala

```diff
@@ -174,7 +175,7 @@ trait ReadOps extends Directives with Messages with Logging {
             case Failure(t: NoDocumentException) =>
                 info(this, s"[GET] entity does not exist")
                 terminate(NotFound)
-            case Failure(t: IllegalArgumentException) =>
```

Comment:
some of the operations e.g. `get` may still throw illegal argument exception because of `requires` in their constructors.